esb/plugins/org.wso2.developerstudio.eclipse.gmf.esb.diagram/src/org/wso2/developerstudio/eclipse/gmf/esb/diagram/debugger/debugpoint/builder/impl/AbstractESBDebugPointBuilder.java

```diff
@@ -0,0 +1,902 @@
+import org.wso2.developerstudio.eclipse.gmf.esb.impl.CloudConnectorOperationImpl;
+import org.wso2.developerstudio.eclipse.gmf.esb.impl.EntitlementAdviceContainerImpl;
+import org.wso2.developerstudio.eclipse.gmf.esb.impl.EntitlementMediatorImpl;
+import org.wso2.developerstudio.eclipse.gmf.esb.impl.EntitlementObligationsContainerImpl;
+import org.wso2.developerstudio.eclipse.gmf.esb.impl.EntitlementOnAcceptContainerImpl;
+import org.wso2.developerstudio.eclipse.gmf.esb.impl.EntitlementOnRejectContainerImpl;
+import org.wso2.developerstudio.eclipse.gmf.esb.impl.FilterFailContainerImpl;
+import org.wso2.developerstudio.eclipse.gmf.esb.impl.FilterMediatorImpl;
+import org.wso2.developerstudio.eclipse.gmf.esb.impl.FilterPassContainerImpl;
+import org.wso2.developerstudio.eclipse.gmf.esb.impl.ForEachMediatorImpl;
+import org.wso2.developerstudio.eclipse.gmf.esb.impl.IterateMediatorImpl;
+import org.wso2.developerstudio.eclipse.gmf.esb.impl.MediatorFlowImpl;
+import org.wso2.developerstudio.eclipse.gmf.esb.impl.MediatorImpl;
+import org.wso2.developerstudio.eclipse.gmf.esb.impl.ProxyServiceImpl;
+import org.wso2.developerstudio.eclipse.gmf.esb.impl.RuleMediatorImpl;
+import org.wso2.developerstudio.eclipse.gmf.esb.impl.SequencesImpl;
+import org.wso2.developerstudio.eclipse.gmf.esb.impl.SwitchCaseContainerImpl;
+import org.wso2.developerstudio.eclipse.gmf.esb.impl.SwitchCaseParentContainerImpl;
+import org.wso2.developerstudio.eclipse.gmf.esb.impl.SwitchDefaultParentContainerImpl;
+import org.wso2.developerstudio.eclipse.gmf.esb.impl.SwitchMediatorImpl;
+import org.wso2.developerstudio.eclipse.gmf.esb.impl.TemplateImpl;
+import org.wso2.developerstudio.eclipse.gmf.esb.impl.ThrottleMediatorImpl;
+import org.wso2.developerstudio.eclipse.gmf.esb.impl.ThrottleOnAcceptContainerImpl;
+import org.wso2.developerstudio.eclipse.gmf.esb.impl.ThrottleOnRejectContainerImpl;
+import org.wso2.developerstudio.eclipse.gmf.esb.impl.ValidateMediatorImpl;
+import org.wso2.developerstudio.eclipse.logging.core.IDeveloperStudioLog;
+import org.wso2.developerstudio.eclipse.logging.core.Logger;
+
+/**
+ * All ESBBreakpoint builders should extend AbstractESBBreakpointBuilder class.
+ * This class contains common methods related to ESBBreakpointBuilders and
+ * implements {@link IESBDebugPointBuilder}
+ *
+ */
+public abstract class AbstractESBDebugPointBuilder implements IESBDebugPointBuilder {
+
+    protected static final String EMPTY_STRING = "";
+    protected static final int INDEX_OF_FIRST_ELEMENT = 0;
+    private EObject innerContainerInstance;
+    private EObject innerParentContainer;
+    private Integer switchCasePosition;
+
+    private static IDeveloperStudioLog log = Logger.getLog(Activator.PLUGIN_ID);
+
+    /**
+     * This method returns the InputConnector of the given mediator.
+     * 
+     * @param mediator
+     * @return
+     */
+    public static InputConnector getInputConnector(Mediator mediator) {
```

Comment:
If getInputConnector is only the method that is being returned, then what's the purpose doing a instance of check here? Is there an updated PR for this?